src/main/resources/static/javascript/mainApp/admin.controller.js

```diff
@@ -389,6 +389,38 @@
             $scope.searchForUserGroupingInformation();
             groupingsService.getAdminLists($scope.getAdminListsCallbackOnSuccess, $scope.displayApiErrorModal);
         };
+        function handleGroupingOwnersOnSuccess(res) {
+            $scope.loading = false;
+            $scope.names = [];
+            for (let i = 0; i < res.groupMembers.length; i++) {
+                $scope.names[parseInt(i, 10)] = res.groupMembers[parseInt(i, 10)].name;
+            }
+            $scope.usernames = [];
+            for (let i = 0; i < res.groupMembers.length; i++) {
+                $scope.usernames[parseInt(i, 10)] = res.groupMembers[parseInt(i, 10)].uid;
+            }
+            $scope.uhuids = [];
+            for (let i = 0; i < res.groupMembers.length; i++) {
+                $scope.uhuids[parseInt(i, 10)] = res.groupMembers[parseInt(i, 10)].uhUuid;
+            }
+        }
+
+        function handleGroupingOwnersOnError() {
+            $scope.loading = false;
+        }
+
+        $scope.displayGroupingOwnersModal = function (groupingPath) {
+            $scope.ownersModalGroupingPath = groupingPath;
+            $scope.groupingOwnersModal = $uibModal.open({
+                templateUrl: "modal/groupingOwnersModal",
+                scope: $scope
+            });
+            groupingsService.groupingOwners($scope.ownersModalGroupingPath, handleGroupingOwnersOnSuccess, handleGroupingOwnersOnError);
+            $scope.displayGroupingOwnersModalOnClose = function () {
+                $scope.groupingOwnersModal.close();
+                $scope.ownersModalGroupingPath = "";
+            };
+        };
```

Comment:
Could you write Jasmine tests for handleGroupingOwnersOnSuccess(), handleGroupingOwnersOnError(), and displayGroupingOwnersModal(). You will have to change the handle functions to be a $scope function in order for Jasmine to call them. Also in the $scope.displayGroupingOwnersModal, be sure to check if the correct handle function was called.